### ISSUE TITLE
Retrieve non-best weights from training results

### ISSUE THREAD

**Issue by MattAlanWright** [ultralytics/ultralytics#2479]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained my own YOLOv8n network using the following command:
```
yolo detect train data=coco128.yaml model=yolov8n.yaml epochs=100000 imgsz=640 show_conf=True save=True save_period=1 project=yolo_training device=0 patience=500
```
You might notice the epochs and patience are both very (probably unecessarily) high. I now have a folder 'weights' filled with .pt models named 'epoch1.pt', 'epoch2.pt', etc, each ~25MB. I also have 'best.pt' and 'last.pt', each only ~6MB. I have 2 questions:

1. What is in the additional ~19MB in the other .pt files and, more importantly, can I use these weights if I would prefer to use the results of a different epoch?
2. Using TensorBoard I can see the validation loss metrics begin to increase around epoch 5000, but the training algorithm tells me epoch 5452 is the "best" one even though the validation losses have already increased by then. What metric is used to choose the "best" weights and am I on the right track in leaning towards ignoring the "best" model and choosing the model at the minimum of the validation loss plots? That seems like a good real-world test of what is the best set of weights.

**Comment by glenn-jocher** [comment]
@MattAlanWright hi! Regarding your first question, the additional ~19MB in the other .pt files likely contains the model weights from the previous epochs that were stored with the `save_period=1` command. Yes, you can use these weights if you would like to use the results of a different epoch. 

As for your second question, the metric used to choose the "best" weights is likely the one you specified in your YAML file (i.e. `metrics: "loss objectness xy"`). Depending on the metric, the weights chosen as "best" might differ from what you would consider the best model based on visual inspection of the validation loss. However, in general, the model with the minimum validation loss is a good indication of the best set of weights.

**Comment by MattAlanWright** [comment]
Gotchya, thank you!

**Comment by glenn-jocher** [comment]
@MattAlanWright you're welcome! If you have any further questions or concerns with using YOLOv8, please don't hesitate to ask.
